sandboxes/dynamicanalysis/analyze-node.js

```diff
@@ -28,12 +32,122 @@ function install(package) {
+
+  let executionErr = null;
   try {
-    require(package.name);
+    executeModuleCode(pkg.name, mod);
   } catch (e) {
-    console.log(`Failed to import ${package.name}: ${e}`);
-  }  
+    executionErr = e;
+    console.log(`Error while executing ${pkg.name} module code: ${e}`);
+  } finally {
+    executionLogStream.close();
+    // restore default console behaviour
+    redirectConsoleWrite(defaultStdoutWrite, defaultStderrWrite);
+  }
+
+  if (executionErr !== null) {
+    // log to normal console too
+    console.log(`Error while executing ${pkg.name} module code: ${executionErr}`);
+  }
+}
+
+// Adapted from https://stackoverflow.com/a/72326559.
+// Premise is that classes have a non-writable prototype
+function isES6Class(obj) {
+  if (typeof obj !== "function") {
+    return false;
+  }
+
+  const descriptor = Object.getOwnPropertyDescriptor(obj, "prototype");
+  return descriptor && !descriptor.writable;
+}
+
+// Best-effort execution of as much code (functions, classes) of the module code as possible
+function executeModuleCode(name, mod) {
+  console.log(`[module] ${name}`);
+  console.log(mod);
+
+  // How to tell if something is a function vs class:
+  // - node uses complex internal logic in util.inspect() function [1]
+  // - StackOverflow thread discussing how complex this problem is [2]
+  //
+  // Solution uses code from an answer in [2]
+  // [1] https://github.com/nodejs/node/blob/main/lib/internal/util/inspect.js
+  // [2] https://stackoverflow.com/questions/30758961/how-to-check-if-a-variable-is-an-es6-class-declaration/72326559
+  const callableNames = Object.keys(mod).filter((key) => typeof mod[key] === "function");
+  console.log(`[keys] ${callableNames}`);
+
+  // Call all the exported names. If there is a TypeError, it's
+  // probably because it is a class, so we'll try again using new.
+  // NOTE: this is a best-effort approach and there are lots of ways
+  // it can fail. In particular, function arguments are not yet supported
+  // TODO basic support for function arguments
+
+  // https://nodejs.org/api/process.html#event-uncaughtexception
+  // tl;dr this may cause things to break, but we're in a sandbox so we'll do it anyway
+  process.on('uncaughtException', (err, origin) => {
+	  console.log("[uncaught exception]");
+	  console.log(err);
+	  console.log(origin);
+  });
+
+  // https://nodejs.org/api/process.html#event-unhandledrejection
+  process.on('unhandledRejection', (reason, promise) => {
+	  console.log("[unhandled rejection]");
+	  console.log(reason);
+	  console.log(promise);
+  });
+
+  for (const name of callableNames) {
+    try {
+      // TODO call each function or class in a separate thread or sandbox
+      if (isES6Class(mod[name])) {
+        console.log(`[class] ${name}`);
+        const instance = new mod[name]();
+		// TODO call instance methods
+      } else {
+        console.log(`[function] ${name}`);
+        mod[name]();
```

Comment:
Do we need to handle timeouts here too? 

We can do this later, but if so let's perhaps gate this behind a flag.

Reply:
This will need to be added in a later PR, for now we'll merge the code but leave it disabled